src/args.rs

```diff
@@ -1,8 +1,125 @@
-#[derive(Debug)]
-pub struct Args {}
+use clap::Parser;
+
+#[derive(Parser, Debug, Default)]
+#[command(author, version, about, long_about = None, name = "jperf")]
+pub struct Args {
+	/// Application name
+	#[arg(long)]
+	app: Option<String>,
+
+	/// Cloud name
+	#[arg(long)]
+	cloud: Option<String>,
+
+	/// Unit name
+	#[arg(long)]
+	unit: Option<String>,
+
+	/// Model name
+	#[arg(long)]
+	model: Option<String>,		
+
+	/// Display a large set of metrics
+	#[arg(short, long)]
+	verbose: bool,
+
+	/// Optional choice of a single metric to display
+	#[arg(short, long)]
+	metric: Option<String>,	
+	
+	/// Most recent time interval in milliseconds
+	#[arg(long, default_value_t = 1000)]
+	interval: u32,
+}
 
 impl Args {
-    pub fn parse() -> Self {
-        todo!()
-    }
+	#[allow(dead_code)]
+	fn app(&self) -> &Option<String>{
+		&self.app
+	}
+
+	#[allow(dead_code)]
+	fn unit(&self) -> &Option<String>{
+		&self.unit
+	}
+	
+	#[allow(dead_code)]
+	fn model(&self) -> &Option<String>{
+		&self.model
+	}
+
+	#[allow(dead_code)]
+	fn cloud(&self) -> &Option<String>{
+		&self.cloud
+	}
+
+	#[allow(dead_code)]
+	fn metric(&self) -> &Option<String>{
+		&self.metric
+	}
+
+	#[allow(dead_code)]
+	fn is_verbose(&self) -> &bool {
+		&self.verbose
+	}
+	
+	#[allow(dead_code)]
+	fn interval(&self) -> &u32 {
+		&self.interval
+	}
+
+	#[allow(dead_code)]
+	fn entity_as_string(&self, entity : &str) -> String {		
```

Comment:
This function both gets the entity and converts it to a `String`, let's keep this simple and just return a `&str` using the first half of the function. Also it looks like this code is just for testing? If so, it would be better to put it into a separate `impl` block and guard it with a `#[cfg(test)]`:
```rust
#[cfg(test)]
impl Args {
    ...
}
```

Reply:
Makes sense, thank you! 
Now returning an `&Option<String>` from this function, as the methods for different entities also return a `&Option<String>`, and the intention with this function is for it to be a single way to call them by an entity str name basically. 
What do you think?

Reply:
I think this is a step in the right direction, I think it would be cleaner to pass around an `Option<&str>` as the borrowing an optional type here feels strange to me---it feels cleaner to give something that whoever is calling this can own, which may or may not contain a reference to something they don't own. This will of course require the getters to be updated, but for that, [Option::as_deref](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref) should help nicely

Reply:
Done!